saphir/src/server.rs

```diff
@@ -361,41 +362,23 @@ impl Server {
             }
         };
 
-        if let Some(request_timeout_ms) = listener_config.request_timeout_ms {
-            use tokio::time::{timeout, Duration};
-            incoming
-                .for_each_concurrent(None, |client_socket| async {
-                    match client_socket {
-                        Ok(client_socket) => {
-                            let peer_addr = client_socket.peer_addr().ok();
-                            let http_handler = http.serve_connection(client_socket, stack.new_handler(peer_addr));
-                            let f = timeout(Duration::from_millis(request_timeout_ms), http_handler);
-
-                            tokio::spawn(f);
-                        }
-                        Err(e) => {
-                            warn!("incoming connection encountered an error: {}", e);
-                        }
+        incoming
+            .for_each_concurrent(None, |client_socket| async {
+                match client_socket {
+                    Ok(client_socket) => {
+                        let peer_addr = client_socket.peer_addr().ok();
+                        if let Some(timeout_ms) = listener_config.request_timeout_ms {
```

Comment:
The conditional needs to be done before hand, it is still evaluated for each new request which is no better, see the old implementation to fix it properly